lib/fhir/common.js

```diff
@@ -119,6 +110,196 @@ module.exports = exports = (mongo) => {
+      if (property == null) {
+        return callback(new Error('Undefined resource property.'))
+      }
+
+      if (Array.isArray(property)) {
+        for (var n = 0; n < property.length; n++) {
+          if (nextProperty[0] && property[n][nextProperty[0]] != null) {
+            evaluateProperties(property[n][nextProperty[0]], nextProperty.slice(1))
+          } else if (property[n].reference) {
+            promises.push(new Promise((resolve, reject) => {
+              resolveReference(property[n].reference, (err, data) => {
+                if (err) {
+                  reject(err)
+                } else {
+                  resolve(data)
+                }
+              })
+            }))
+          } else {
+            return callback(new Error(`Invalid resource reference: "${JSON.stringify(property[n])}"`))
+          }
+        }
+      } else {
+        if (nextProperty[0] && property[nextProperty[0]] != null) {
+          evaluateProperties(property[nextProperty[0]], nextProperty.slice(1))
+        } else if (property.reference) {
+          promises.push(new Promise((resolve, reject) => {
+            resolveReference(property.reference, (err, data) => {
+              if (err) {
+                reject(err)
+              } else {
+                resolve(data)
+              }
+            })
+          }))
+        } else {
+          return callback(`Invalid resource reference: "${JSON.stringify(property)}"`)
+        }
+      }
+    }
+
+    evaluateProperties(resultList[includeReference[0]], includeReference.slice(1))
+
+    // resolve promises
+    Promise.all(promises)
+      .then((res) => {
+        callback(null, res)
+      }).catch((err) => {
+        callback(err)
+      })
+  }
+
+  const parseSearchExpression = (exp, callback) => {
+    const itemList = exp.split(':')
+    const itemResource = itemList[0]
+    const itemReference = itemList[1].split('.')
+
+    callback(itemResource, itemReference)
+  }
+
+  const resolveIncludeParameters = (item, results) => {
+    return new Promise((resolve, reject) => {
+      const promises = []
+
+      const itemList = item.split('.')
+      const itemResource = itemList[0]
+      const itemReference = itemList.slice(1)
+
+      results.forEach((item) => {
+        if (item['resourceType'] === itemResource) {
+          promises.push(new Promise((resolve, reject) => {
+            resolveNestedProperties(item, itemReference, (err, data) => {
+              if (err) {
+                reject(err)
+              } else {
+                resolve(data)
+              }
+            })
+          }))
+        }
+      })
+
+      Promise.all(promises)
+        .then((res) => {
+          const flattenedRes = [].concat.apply([], res)
+
+          resolve(flattenedRes)
+        })
+        .catch((err) => {
+          reject(err)
+        })
+    })
+  }
+
+  function mapSearchNameToPath (searchParameters, callback) {
+    const promises = []
+    if (searchParameters) {
```

Comment:
Check that search `searchParameters` are empty and exit early instead of branching your code. 
```javascript
if(searchParameters == null){
    return callback('No _include search parameters supplied.')
}

//Continue with your function
```